### PR TITLE
removing `version-tags` from public image repo

### PR DESCRIPTION
**Pull request by imjasonh**
This begins to remove version tagging infrastructure from public image modules. These have been effectively disabled for nearly all images since Aug 16, via the env var. This removes the scaffolding that tries to version tag things at all. 🧹

### CODE REVIEW

File: images/argocd/main.tf [hunk 1]
```diff
@@ -15,6 +21,26 @@ module "latest" {
...
+}
+
+resource "oci_tag" "latest-dev" {
+  for_each = module.latest
+
+  digest_ref = each.value.dev_ref
+  tag        = "latest-dev"
+}
+
+// TODO: Remove this.
```
Comment (mattmoor) [inline images/argocd/main.tf:43]: why wait?

File: images/aws-cli/main.tf [hunk 2]
```diff
@@ -10,24 +16,38 @@ module "latest" {
...
+  tag        = "latest"
+}
+
+resource "oci_tag" "latest-dev" {
+  depends_on = [module.test-latest]
+  digest_ref = module.latest.dev_ref
+  tag        = "latest-dev"
+}
+
+// TODO: Remove this.
```
Comment (mattmoor) [inline images/aws-cli/main.tf:36]: Why wait?

**Review by mattmoor — COMMENTED** [review]
This all LGTM, why not remove the stuff with TODOs now other than it taking time to complete the other PR after this lands?

**Comment by imjasonh** [comment]
Removed the things.

**Review by mattmoor — APPROVED** [review]

**Pull request merged**